### DISCUSSION TITLE
Is it possible to create .msh file using only T1?

### DISCUSSION THREAD

**Discussion by wustew** [simnibs/simnibs#160] [7.Creating head models] [answered]
Hello, tutorial states that T2 is optional when using charm to build head model. However, when I just use T1, I do not get a .msh file output.  Should I be using both T1 and T2? Thanks.

**Comment by oulap** [comment]
charm doesn't require a T2, so it sounds like something went wrong with the run. Do you have the charm_log.html? It should be inside the m2m-folder.

**Reply by wustew** [reply]
Here is the charm_log.  I cut out some of iteration log to decrease file output.  Thank you.

[charm_log.txt](https://github.com/simnibs/simnibs/files/11925669/charm_log.txt)

**Reply by jduemose** [reply, marked as answer]
It seems you are getting an error from nibabel (which used to be only a warning: `get_data` vs `get_fdata`). Perhaps you can check the version of nibabel (e.g., `simnibs_python -c "import nibabel; print(nibabel.__version__)"`). Did you update `nibabel` yourself? I guess that since the first run fails to write the MNI warps etc. that is what causes the read error in the `--segment` run.

If nibabel prints version >5 perhaps try to downgrade to version 3 or 4 or download the newest simnibs release (4.0.1). I believe I fixed these deprecated calls in this version.

Jesper

**Reply by wustew** [reply]
I manually updated nibabel for another task. I deleted simnibs v4.0.0 and installed v4.0.1 and mesh construction was completed.  Thank you.
